x/dsmr/node_test.go

```diff
@@ -29,6 +32,26 @@ var (
 	_ Verifier[tx] = (*failVerifier)(nil)
 )
 
+func newTestingLog() logging.Logger {
+	return logging.NewLogger("dsmr")
+}
+
+func newTestingTracer(t *testing.T) trace.Tracer {
+	tracer, err := trace.New(trace.Config{})
+	require.NoError(t, err)
+	return tracer
+}
+
+type testingChainIndex struct{}
+
+func (*testingChainIndex) GetExecutionBlock(context.Context, ids.ID) (validitywindow.ExecutionBlock[*ChunkCertificate], error) {
+	return nil, nil
+}
+
+func newChainIndexer() ChainIndex {
+	return &testingChainIndex{}
+}
```

Comment:
We should add unit tests for the newly added functionality here. What do you think of the following cases?

- skip duplicate chunks in build block
- return an error if all available chunks are duplicates in build block
- duplicate chunk within a block
- duplicate chunk that between the block we are executing and the last accepted block
- duplicate chunk inside of the accepted validity window
- propagate an error if we fail to fetch a block during de-duplication

Reply:
done : unit test for `skip duplicate chunks in build block`

Reply:
done - add cases added.